MaterialLib/Component/Dispersivity/createDispersivity.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file:   createDispersivity.cpp
+ */
+
+#include "createDispersivity.h"
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+#include "Dispersivity.h"
+
+namespace MaterialLib
+{
+namespace Component
+{
+std::unique_ptr<Dispersivity> createDispersivity(
+    BaseLib::ConfigTree const& config,
+    std::vector<std::unique_ptr<ProcessLib::ParameterBase>> const& parameters)
+{
+    //! \ogs_file_param{material__component__dispersivity__type}
+    auto const type = config.getConfigParameter<std::string>("type");
+
+    if (type == "Constant")
+    {
+        auto const& constant_dispersivity = ProcessLib::findParameter<double>(
+            config,
+            //! \ogs_file_param_special{material__component__dispersivity__dispersivity_parameters}
+            "dispersivity_parameters", parameters, 2);
+
+        return std::make_unique<Dispersivity>(constant_dispersivity);
+    }
+
+    OGS_FATAL("The dispersivity type %s is unavailable.\n",
+              "The available type is Constant.",
+              type.data());
```

Comment:
I think its better to use `type.c_str()` that gives you the non-modifiable standard C character array version of the string.

Reply:
Corrected.